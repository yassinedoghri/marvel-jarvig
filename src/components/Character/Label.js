import styled from "styled-components";

const Label = styled.label`
  display: inline-block;
  position: relative;
  padding: ${props => props.theme.spaces.md}em;
`;

export default Label;
