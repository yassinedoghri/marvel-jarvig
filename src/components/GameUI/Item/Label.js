import styled from "styled-components";


const Label = styled.span`
    font-family: Bangers;
    letter-spacing: .1em;
    text-shadow:
       3px 3px 0 #222,
     -1px -1px 0 #222,  
      1px -1px 0 #222,
      -1px 1px 0 #222,
       1px 1px 0 #222;
`;

export default Label;
