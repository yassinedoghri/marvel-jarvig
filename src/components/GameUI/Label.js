import { flash } from "react-animations";
import styled, { keyframes } from "styled-components";

const flashAnimation = keyframes`${flash}`;

const Label = styled.span`
  font-family: Bangers, sans-serif;
  letter-spacing: 0.1em;
  text-shadow: 3px 3px 0 #222, -1px -1px 0 #222, 1px -1px 0 #222,
    -1px 1px 0 #222, 1px 1px 0 #222;
  animation: ${props =>
    props.blink ? `2s ${flashAnimation} infinite` : "none"};
`;

export default Label;
