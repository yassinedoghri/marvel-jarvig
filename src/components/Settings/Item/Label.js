import styled from "styled-components";

const Label = styled.p`
    display: inline-block;
    margin-bottom: .5em;
`;

export default Label;
