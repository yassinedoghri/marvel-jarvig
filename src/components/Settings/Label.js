import styled from "styled-components";

const Label = styled.p`
  display: inline-block;
  margin-bottom: ${props => props.theme.spaces.sm}em;
`;

export default Label;
