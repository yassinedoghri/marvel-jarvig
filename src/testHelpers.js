import React from "react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import theme from "theme";
import { ThemeProvider } from "styled-components";

export default function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}
