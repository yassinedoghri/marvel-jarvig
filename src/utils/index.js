export { default as style } from "utils/style";
export { default as baseStyles } from "utils/baseStyles";
export { default as helpers } from "utils/helpers";
export { default as testHelpers } from "utils/testHelpers";
export { default as registerServiceWorker } from "utils/registerServiceWorker";
export { default as media } from "utils/media";
