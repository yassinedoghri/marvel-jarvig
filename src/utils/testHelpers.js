import { createMemoryHistory } from "history";
import React from "react";
import { Provider } from "react-redux";
import {
  ConnectedRouter,
  routerMiddleware,
  routerReducer
} from "react-router-redux";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import rootReducer from "reducers";
import { applyMiddleware, combineReducers, createStore } from "redux";
import { ThemeProvider } from "styled-components";
import theme from "theme";

const history = createMemoryHistory("/");
const routeMiddleware = routerMiddleware(history);

const store = createStore(
  combineReducers({
    ...rootReducer,
    router: routerReducer
  }),
  applyMiddleware(routeMiddleware)
);

export function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export function renderInContext(component) {
  return renderer.create(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}

export function ShallowRenderInContext(component) {
  const renderer = new ShallowRenderer();
  return renderer.render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}
