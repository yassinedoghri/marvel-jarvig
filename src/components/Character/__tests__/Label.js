import Label from "components/Character/Label";
import React from "react";
import { renderWithTheme } from "utils/testHelpers";

describe("Character Label", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(<Label />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
