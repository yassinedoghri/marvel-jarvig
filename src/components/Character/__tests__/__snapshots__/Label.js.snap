// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Label renders correctly 1`] = `
<label
  className="sc-bdVaJa gZaPKc"
/>
`;
